### PR TITLE
Add filter and search definitions

### LINKED ISSUE #3312: Search vs Filter added to https://www.patternfly.org/v4/ux-writing/terminology/
Filter vs search seems nuanced, and like something we should consider adding more context around for users. Is this laid out somewhere already that I couldn't find?  

From https://www.patternfly.org/v4/guidelines/filters -  Filters allow users to narrow down content from data in tables, data lists or card views, among many others. 
The wires here say Filter.

From https://www.patternfly.org/v4/components/search-input/design-guidelines/#search-input-as-a-type-ahead-filter - A search input is a type of input field that can be user (there's a typo here, should be **used**) to search, find, or filter. See filter guidelines for more information on using search input as a filter.
The demos here either say Find or have no text in them.

I'd suggest updating https://www.patternfly.org/v4/ux-writing/terminology/ to include filter vs search to the "list of common UI terms and their usage."

### PR DESCRIPTION
**Pull request by mcarrano**
Add definitions for the terms Filter and Search to the UX-writing/Terminology page to address this issue: #3312

Closes #3312

### CODE REVIEW

File: packages/v4/patternfly-docs/content/design-guidelines/content/terminology.md [hunk 1]
```diff
@@ -80,6 +82,8 @@ _For Red Hat product terminology and documentation standards, see [Red Hat's sup
 **Save**: (v.) Use to describe preserving changes made by the user. 
 - Opposite: Cancel
 
+**Search** (v.) Use to decribe the action of querying a data set to return a results list. Searching will always be applied against some criteria and may be narrow or broard.
```
Comment (abigaeljamie) [inline packages/v4/patternfly-docs/content/design-guidelines/content/terminology.md:85]:
```suggestion
**Search** (v.) Use to decribe the action of querying a data set to return a results list. Searching will always be applied against some criteria and may be narrow or broad. 
```  
Small typo, changed "broard" to "broad"

**Review by abigaeljamie — CHANGES_REQUESTED** [review]

**Comment by mcarrano** [comment]
@abigaeljamie @mmenestr I updated the Terminology page in the UX writing section to add definitions for Filter and Search as referenced in the issue. Please give it a look here: https://patternfly-org-pr-3334-v4.surge.sh/v4/ux-writing/terminology

**Comment by abigaeljamie** [comment]
> @abigaeljamie @mmenestr I updated the Terminology page in the UX writing section to add definitions for Filter and Search as referenced in the issue. Please give it a look here: https://patternfly-org-pr-3334-v4.surge.sh/v4/ux-writing/terminology

Thanks, Matt! I submitted a review. I found 1 typo, but otherwise the terms look great (thank you for following the terms structure to a tee!).

File: packages/v4/patternfly-docs/content/design-guidelines/content/terminology.md [hunk 2]
```diff
@@ -45,6 +45,8 @@ _For Red Hat product terminology and documentation standards, see [Red Hat's sup
 **Expand**: (v.) Use to describe expanding a container element (like a list or message) to show all its contents.
 - Opposite: Collapse
 
+**Filter**: (v.) Use to describe the action of narrowing a set of search results to show only items meeting a certain criteria. Filtering is mostly associate with card views like tables, lists, and card grids and assumes that a search has previously been performed to return an initial results list.
```
[the remark this answers is not in the document]
Comment (mmenestr) [reply packages/v4/patternfly-docs/content/design-guidelines/content/terminology.md:48]: Also maybe it's just me, but calling tables and lists types of "card views" seem odd to me... wondering if we could use a different word?

[the remark this answers is not in the document]
Comment (mcarrano) [reply packages/v4/patternfly-docs/content/design-guidelines/content/terminology.md:48]: Yes, I think that is a typo

**Comment by mcarrano** [comment]
@abigaeljamie @mmenestr I made the changes that you suggested. Can you review this one more time? Thanks.

**Review by mmenestr — APPROVED** [review]
LGTM!

**Comment by mmenestr** [comment]
Once @abigaeljamie approves, I will merge !

**Review by abigaeljamie — APPROVED** [review]

**Pull request merged**